lib/Doctrine/DBAL/Driver/PDOPgSql/Driver.php

```diff
@@ -53,6 +53,14 @@ public function connect(array $params, $username = null, $password = null, array
                 $pdo->setAttribute(PDO::PGSQL_ATTR_DISABLE_PREPARES, true);
             }
 
+            /* defining client_encoding via SET NAMES to avoid inconsistent DSN support
+             * - the 'client_encoding' connection param only works with postgres >= 9.1
+             * - passing client_encoding via the 'options' param breaks pgbouncer support
+             */
+            if (isset($params['charset'])) {
+              $pdo->query('SET NAMES \''.$params['charset'].'\'');
```

Comment:
Yeah I'm def. not confused. @zeroedin-bill those suggestions aren't valid unfortunately.

IE, PGSQL_LIBPQ_VERSION gives me 8.4 on my web server (default version from a centos 6 install) -- but of course I have 9.3 on my db server. Most of us aren't running on a single-server environment!

The first one out of the three is the only one that would actually work.

@deeky666 what do you think about the suggestion of trying with client_encoding, and falling back to SET NAMES on failure? I can make those modifications easy enough. That way -- you only have the double connection if you're using an old (unsupported mind you) version of PG.